banksystem/users/models.py

```diff
@@ -1,12 +1,36 @@
 from django.db import models
+from django.contrib.auth.models import BaseUserManager, AbstractUser
 
 
-class User(models.Model):
-    first_name = models.CharField(max_length=100)
-    last_name = models.CharField(max_length=100)
-    username = models.CharField(max_length=300)
-    email = models.EmailField()
-    password = models.CharField(max_length=100)
-    password_confirmation = models.CharField(max_length=100)
+class CustomUserManager(BaseUserManager):
+    def create_user(self, email, password=None, **extra_fields):
+        if not email:
+            raise ValueError("Email field is required.")
+        email = self.normalize_email(email)
+        user = self.model(email=email, **extra_fields)
+        user.set_password(password)
+        user.save(using=self._db)
+        return user
+
+    def create_superuser(self, email, password=None, **extra_fields):
+        extra_fields.setdefault('is_staff', True)
+        extra_fields.setdefault('is_superuser', True)
+        return self.create_user(email, password, **extra_fields)
+
+
+class CustomUser(AbstractUser):
+    email = models.EmailField(unique=True)
+    username = models.CharField(max_length=150, unique=True)
+    firstname = models.CharField(max_length=150)
+    lastname = models.CharField(max_length=150)
+    password = models.CharField(max_length=150)
+    password_confirmation = models.CharField(max_length=150)
```

Comment:
Remove the fields which are already declared in the parent classes, and only declare the fields here which are new